conda/mpich/build.sh

```diff
@@ -1,99 +1,13 @@
-
-unset LDFLAGS CPPFLAGS CFLAGS CXXFLAGS FFLAGS FCFLAGS F90 F77
-export LDFLAGS="-L$PREFIX/lib -Wl,-rpath,$PREFIX/lib"
-export LIBRARY_PATH="$PREFIX/lib"
-
-if [[ $(uname) == Darwin ]]; then
-    if [[ $target_platform == osx-arm64 ]]; then
-        CTUNING="-mcpu=apple-a12 -I$PREFIX/include"
-        FTUNING="-march=armv8.3-a -I$PREFIX/include"
-        OPTIONS="--disable-opencl --enable-cxx --enable-fortran"
-        export pac_cv_f77_accepts_F=yes
-        export pac_cv_f77_flibs_valid=unknown
-        export pac_cv_f77_sizeof_double_precision=8
-        export pac_cv_f77_sizeof_integer=4
-        export pac_cv_f77_sizeof_real=4
-        export pac_cv_fc_accepts_F90=yes
-        export pac_cv_fc_and_f77=yes
-        export pac_cv_fc_module_case=lower
-        export pac_cv_fc_module_ext=mod
-        export pac_cv_fc_module_incflag=-I
-        export pac_cv_fc_module_outflag=-J
-        export pac_cv_fort90_real8=yes
-        export pac_cv_fort_integer16=yes
-        export pac_cv_fort_integer1=yes
-        export pac_cv_fort_integer2=yes
-        export pac_cv_fort_integer4=yes
-        export pac_cv_fort_integer8=yes
-        export pac_cv_fort_real16=no
-        export pac_cv_fort_real4=yes
-        export pac_cv_fort_real8=yes
-        export pac_cv_prog_f77_and_c_stdio_libs=none
-        export pac_cv_prog_f77_exclaim_comments=yes
-        export pac_cv_prog_f77_has_incdir=-I
-        export pac_cv_prog_f77_library_dir_flag=-L
-        export pac_cv_prog_f77_mismatched_args=yes
-        export pac_cv_prog_f77_mismatched_args_parm=
-        export pac_cv_prog_f77_name_mangle='lower uscore'
-        export pac_cv_prog_fc_and_c_stdio_libs=none
-        export pac_cv_prog_fc_int_kind_16=8
-        export pac_cv_prog_fc_int_kind_8=4
-        export pac_cv_prog_fc_works=yes
-        export pac_MOD='mod'
-    else
-        CTUNING="-march=core2 -mtune=haswell -I$PREFIX/include"
-        FTUNING="-march=core2 -mtune=haswell -I$PREFIX/include"
-        OPTIONS=""
-    fi
-    SHARED="clang"
-else
-    SHARED="gcc"
-    CTUNING="-march=nocona -mtune=haswell -I$PREFIX/include"
-    FTUNING="-march=nocona -mtune=haswell -I$PREFIX/include"
-    OPTIONS=""
-fi
-
-./configure --prefix="${PREFIX}" \
-            --enable-shared \
-            --enable-sharedlibs="${SHARED}" \
-            --enable-fast=O2 \
-            --enable-debuginfo \
-            --enable-two-level-namespace \
-            --with-device=ch3 \
-            CC="${CC}" CXX="${CXX}" FC="${FC}" F77="${FC}" F90="" \
-            CFLAGS="${CTUNING}" CXXFLAGS="${CTUNING}" FFLAGS="${FTUNING}" LDFLAGS="${LDFLAGS}" \
-            FCFLAGS="${FTUNING}" F90FLAGS="" F77FLAGS="" \
-            ${OPTIONS}
-CORES=$(echo "${CPU_COUNT:-2} / 2" | bc)
-make -j $CORES
-make install
+TEMP_CXXFLAGS=${CXXFLAGS%%-fdebug-prefix-map*}-std=c++17
+# Finally, swap "-std=c++14" with "" to set our basic standard requirement.
+ACTIVATION_CXXFLAGS=${TEMP_CXXFLAGS/-std=c++14/}
```

Comment:
Either line 10 is wrong or the comment is wrong

Reply:
The core of the comment is correct - this does substitute C++14 if it exists with an empty string. But the modified comment could be improved. I'll fix in the next round of commits.

Reply:
To future proof this, we should probably do

```
${TMP_CXXFLAGS//-std=c++[0-9]{2}/}
```

to replace _all_ `-std=c++xx` with nothing

Reply:
Oh I like that better. Will do.

Reply:
Adjusted a little, as the regex is slightly different than the standard bash pattern matching that this substitution expects, but fixed up in a revised [1cc5c0a](https://github.com/idaholab/moose/pull/20580/commits/1cc5c0af8a5d0385f03d696710d9aeecae130a18)